Automattic/wp-calypso
# Automattic Components: Add select dropdown to package

- Dependent on https://github.com/Automattic/wp-calypso/pull/79257 and https://github.com/Automattic/wp-calypso/pull/79256
- Tracking issue https://github.com/Automattic/martech/issues/2023

## Proposed Changes

I understand that these seem like a large volume of changes, but we're simply cloning the existing `client/components/` select dropdown component into `@automattic/components` as we did in https://github.com/Automattic/wp-calypso/pull/79257 and https://github.com/Automattic/wp-calypso/pull/79256

The entirety of the migration is a 4 step process that creates easily digestible ( and revertible ) PRs to work with. See https://github.com/Automattic/martech/issues/2028. Unfortunately, this process also eliminates the git history from migrated files unless the 4 steps are condensed into a single one. I don't have a great answer to this, but the tradeoff, to me, is worth the added safety of more understandable, compact, iterative pull requests.

## GIFs
![2023-08-12 20 49 14](https://github.com/Automattic/wp-calypso/assets/5414230/086fce28-fa20-4486-8d6f-7cf69466c786)

## Testing Instructions



* Make sure all tests pass. `yarn run test-packages packages/components/src/select-dropdown/test` if you'd like to run it locally, but the CI process should catch anything.
* At the root of the Calypso directory, run `yarn storybook:start`
* Ensure that the select-dropdown component renders properly

## Pre-merge Checklist
